ksql-parser/src/test/java/io/confluent/ksql/schema/ksql/LogicalSchemasTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.parser.tree.PrimitiveType;
+import io.confluent.ksql.parser.tree.Type;
+import io.confluent.ksql.parser.tree.Type.SqlType;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class LogicalSchemasTest {
+
+  private static final Schema LOGICAL_BOOLEAN_SCHEMA = SchemaBuilder.bool().optional().build();
+  private static final Schema LOGICAL_INT_SCHEMA = SchemaBuilder.int32().optional().build();
+  private static final Schema LOGICAL_BIGINT_SCHEMA = SchemaBuilder.int64().optional().build();
+  private static final Schema LOGICAL_DOUBLE_SCHEMA = SchemaBuilder.float64().optional().build();
+  private static final Schema LOGICAL_STRING_SCHEMA = SchemaBuilder.string().optional().build();
+
+  private static final java.util.Map<Type, Schema> SQL_TO_LOGICAL = ImmutableMap.<Type, Schema>builder()
+      .put(PrimitiveType.of(SqlType.BOOLEAN), LOGICAL_BOOLEAN_SCHEMA)
+      .put(PrimitiveType.of(SqlType.INTEGER), LOGICAL_INT_SCHEMA)
+      .put(PrimitiveType.of(SqlType.BIGINT), LOGICAL_BIGINT_SCHEMA)
+      .put(PrimitiveType.of(SqlType.DOUBLE), LOGICAL_DOUBLE_SCHEMA)
+      .put(PrimitiveType.of(SqlType.STRING), LOGICAL_STRING_SCHEMA)
+      .put(io.confluent.ksql.parser.tree.Array.of(PrimitiveType.of(SqlType.INTEGER)),
+          SchemaBuilder.array(LogicalSchemas.INTEGER).optional().build())
+      .put(io.confluent.ksql.parser.tree.Map.of(PrimitiveType.of(SqlType.INTEGER)),
+          SchemaBuilder.map(LogicalSchemas.STRING, LogicalSchemas.INTEGER).optional().build())
+      .put(io.confluent.ksql.parser.tree.Struct.builder()
+              .addField("f0", PrimitiveType.of(SqlType.INTEGER))
+              .build(),
+          SchemaBuilder.struct().field("f0", LogicalSchemas.INTEGER).optional().build())
+      .build();
+
+  private static final java.util.Map<Schema, Type> LOGICAL_TO_SQL = ImmutableMap.copyOf(
```

Comment:
you can use `ImmutableBiMap` if you are so inclined

Reply:
Nice.